### PR TITLE
Add verbose levels

### LINKED ISSUE #36: Better verbosity options
Improve verbosity settings for DiffKemp. Enable multiple verbosity levels controlling various outputs (info about comparison results, builder output, various debugging information). This should be also implemented for SimpLL (introduce a verbosity CLI option instead of using `#ifdef DEBUG`).

### PR DESCRIPTION
**Pull request by Kuba314**
Added a new, more verbose verbosity level.

This new verbosity level can be selected by using a `-vv` parameter.
So far I've only moved each passes' debug output to this new level. Other debug messages stay on the same level as before.

Resolves #36

### CODE REVIEW

**Review by lenticularis39 — CHANGES_REQUESTED** [review]
Please have a look at the code style CI run, there are some complaints from clang-format about code formatting.

File: diffkemp/simpll/Config.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include <llvm/Support/SourceMgr.h>
 
 #define DEBUG_SIMPLL "debug-simpll"
+#define VERBOSE_SIMPLL "verbose-simpll"
```
Comment (viktormalik) [inline diffkemp/simpll/Config.h:24]: `DEBUG_SIMPLL_VERBOSE` would be better here (to be consistent with the other debug types by using the `DEBUG` prefix).

File: diffkemp/simpll/Config.cpp [hunk 2]
```diff
@@ -161,9 +169,17 @@ Config::Config(std::string FirstFunName,
     }
 
     std::vector<std::string> debugTypes;
-    if (Verbose) {
-        // Enable debugging output in passes
-        debugTypes.emplace_back(DEBUG_SIMPLL);
+    if (VerbosityOpt > 0) {
```
Comment (viktormalik) [inline diffkemp/simpll/Config.cpp:172]: Should be like this, I believe:
```suggestion
    if (Verbosity > 0) {
```
`VerbosityOpt` is not relevant here as this constructor is used when SimpLL is **not** invoked from CLI.

File: diffkemp/simpll/Config.cpp [hunk 3]
```diff
@@ -161,9 +169,17 @@ Config::Config(std::string FirstFunName,
...
+
+        // Enable debugging output in passes. Intended fallthrough
+        switch (VerbosityOpt) {
```
Comment (viktormalik) [inline diffkemp/simpll/Config.cpp:175]:
```suggestion
        switch (Verbosity) {
```

**Review by viktormalik — CHANGES_REQUESTED** [review]
I think that we should eventually merge the `--verbose-macros` option with the new verbosity levels. There are two options:
- add it to level 2
- create a new level for it (3)

I have a slight preference towards the second option as the output with `--verbose-macros` is quite long. WDYT @lenticularis39?

Anyways, I'm completely fine with doing this in a separate PR, just mentioning it here as it is related.

File: diffkemp/simpll/Config.h (hunk 1 above)
Comment (Kuba314) [reply diffkemp/simpll/Config.h:24]: What would be the 3rd level called? `DEBUG_SIMPLL_VERBOSE_MACROS` or `DEBUG_SIMPLL_VERBOSE_3`. I don't think either of those would be fitting. The first one says that only macros are at this level, and the third one wouldn't quite fit the naming scheme.

Comment (viktormalik) [reply diffkemp/simpll/Config.h:24]: We could call it `DEBUG_SIMPLL_VERBOSE_EXTRA`, that way, each level has an added word.

**Comment by Kuba314** [comment]
- Added third verbosity level (`-vvv`)
- Fixed code formatting
- Rebased onto master

**Comment by viktormalik** [comment]
Oh, and please respect the 50/72 rule for the commit messages.

**Review by viktormalik — APPROVED** [review]

**Comment by viktormalik** [comment]
@lenticularis39 could you please look at this? Thanks.

File: diffkemp/simpll/Config.cpp [hunk 4]
```diff
@@ -107,13 +104,19 @@ Config::Config()
...
-        debugTypes.emplace_back(DEBUG_SIMPLL_MACROS);
+    if (VerbosityOpt > 0) {
+
+        // Enable debugging output in passes. Intended fallthrough
+        switch (VerbosityOpt) {
+        default:
+        case 3:
+            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE_EXTRA);
+        case 2:
+            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE);
```
Comment (lenticularis39) [inline diffkemp/simpll/Config.cpp:115]: `[[fallthrough]]` attribute can be used to tell the compiler that the fallthough is intented: 
```suggestion
        case 3:
            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE_EXTRA);
            [[fallthrough]];
        case 2:
            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE);
            [[fallthrough]];
```

File: diffkemp/simpll/Config.cpp [hunk 5]
```diff
@@ -161,13 +163,19 @@ Config::Config(std::string FirstFunName,
...
+        switch (Verbosity) {
+        default:
+        case 3:
+            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE_EXTRA);
+        case 2:
+            debugTypes.emplace_back(DEBUG_SIMPLL_VERBOSE);
+        case 1:
+            debugTypes.emplace_back(DEBUG_SIMPLL);
+            break;
+        }
```
Comment (lenticularis39) [inline diffkemp/simpll/Config.cpp:178]: The comments from the first occurrence of this code block also apply here.

**Review by lenticularis39 — APPROVED** [review]

**Pull request merged**